vaadin-date-picker-flow-parent/vaadin-date-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datepicker/DatePickerLocaleIT.java

```diff
@@ -20,79 +19,97 @@
 @TestPath("vaadin-date-picker/date-picker-locale")
 public class DatePickerLocaleIT extends AbstractComponentIT {
 
-    @Test
-    public void testPickerWithValueAndLocaleFromServerSideDifferentCtor() {
+    @Before
+    public void init() {
         open();
+    }
 
-        DatePickerElement localePicker = $(DatePickerElement.class)
-                .id("locale-picker-server-with-value");
-        WebElement displayText = localePicker.findElement(By.tagName("input"));
-
-        Assert.assertEquals("Wrong initial date in field.", "2018/4/23",
-                executeScript("return arguments[0].value", displayText));
+    @Test
+    public void datePickerWithValueAndLocale_assertDisplayedValue() {
+        DatePickerElement frenchLocalePicker = $(DatePickerElement.class)
+                .id("picker-with-value-and-french-locale");
+        Assert.assertEquals("Should display the value using French date format",
+                "03/05/2018", frenchLocalePicker.getInputValue());
+
+        DatePickerElement koreanLocalePicker = $(DatePickerElement.class)
+                .id("picker-with-value-and-korean-locale");
+        Assert.assertEquals("Should display the value using Korean date format",
+                "2018. 5. 3.", koreanLocalePicker.getInputValue());
+
+        DatePickerElement polishLocalePicker = $(DatePickerElement.class)
+                .id("picker-with-value-and-polish-locale");
+        Assert.assertEquals("Should display the value using Polish date format",
+                "3.05.2018", polishLocalePicker.getInputValue());
+
+        assertNoWarnings();
+    }
 
-        findElement(By.id("uk-locale")).click();
-        Assert.assertEquals("Didn't have expected UK locale date.",
-                "23/04/2018",
-                executeScript("return arguments[0].value", displayText));
```

Comment:
**note:** Switching to UK locale is covered by https://github.com/vaadin/flow-components/pull/3643/files#diff-d62e2a4757951e946bd0e831e8193bb3f375145b9bc7b4b4ce2bf9e969bfc938R74